### DISCUSSION TITLE
Re-embedding memory when model changes

### DISCUSSION THREAD

**Discussion by Fusseldieb** [open-webui/open-webui#7604] [Ideas]
# Feature Request

![image](https://github.com/user-attachments/assets/a49efc61-0a5a-4ef7-a195-5d9ee2d97e7b)

**Is your feature request related to a problem? Please describe.**
Yes, if we change the embedding model to any other, the embedding vectors themselves get lost, and there is no way to recreate them unless we delete and re-add ***all memories***, which are a lot in my case.

**Describe the solution you'd like**
A button that goes through each memory and re-embedds it. So, if the vector database got wiped, this button would re-embed everything. In fact, this would ideally make more sense to be a global option that re-embeds it for every user.

**Describe alternatives you've considered**
Restarting OpenWebUI, changing embedding models, etc. Only "Updating" a memory re-creates the embedding for that particular memory.

**Additional context**
\-

**Comment by flefevre** [comment]
Additionally I think it could be pertinent to attache the documents configuration from general admin panel to each knowledge base.
Every document data set have a specific embedding model or parameters ( chunk or overlap or template rag)

**Reply by ChromaticPanic** [reply]
This is how RAGFlow does it and it makes sense. RAGFlow also has a button to reparse single files or the whole folder. It would be good if OpenWebUI had that, even if just in the api so we can script it.

**Comment by mahenning** [comment]
It seems there is an API call for that already in the code:
https://github.com/open-webui/open-webui/blob/63533c9e3ab41edd7bd4124ef94f6b6dc09aa175/backend/open_webui/routers/memories.py#L94-L122

I think "reset" is not the best wording for it, but looking at the code:
- it takes all memories from the current user and deletes the embeddings from the `VECTOR_DB_CLIENT` (line 101)
- then gets the user memory from the database (not the same as vector db) (line 103)
- re-embedds each memory into the vector db (line 104+)

I didn't test the endpoint yet, but it looks pretty much like it's doing what you asked, for one user. Would be nice to be call-able in the UI though..

**Reply by mahenning** [reply]
In case anyone wants to run it, open a terminal and insert:
> curl -X POST -H "Authorization: Bearer sk-\<your-api-key\>" -H "Content-Type: application/json" http://localhost:3000/api/v1/memories/reset

If your Open-Webui doesn't run under "localhost:3000", you have to change that. You also have to create your own open-webui api key to insert into the curl command. I didn't test it with different embedding models, but I can connfirm that the memories of the current user are re-embedded.

**Comment by ips972** [comment]
This was available in old version of open webui, in the documents settings , when you made any changes you had a button that reran the embedding of documents in the download directory or something like that. At some point is was removed. 

I have this same problem since I’m thinking about switching embedding model or replacing the vectorDB - and all the knowledge and personal data - can’t be simply recalculated and embedded.. I need to delete it all and recreate.

**Comment by i-iooi-i** [comment]
I've also run into the same issue.

**Comment by mahenning** [comment]
I have a PR (https://github.com/open-webui/open-webui/pull/13225) to fix that for files, knowledge and memories, that's almost ready 👍  test and feedback is appreciated.

**Comment by guenhter** [comment]
Would love to see this
